File: cpp/epi/search_in_bst.cc
#include <memory>

#include "bst_node.h"
#include "generic_test.h"
using std::unique_ptr;

BstNode<int>* SearchBST(const unique_ptr<BstNode<int>>& tree, int key) {
  return nullptr;
}

int SearchBSTWrapper(const unique_ptr<BstNode<int>>& tree, int key) {
  auto result = SearchBST(tree, key);
  return result ? result->data : -1;
}

int main(int argc, char* argv[]) {
  std::vector<std::string> args{argv + 1, argv + argc};
  std::string func_name = "SearchBST";
  std::vector<std::string> param_names{"tree", "key"};

  return GenericTestMain(args, "search_in_bst.cc",
                         "../test_data/epi/search_in_bst.tsv",
                         &SearchBSTWrapper, DefaultComparator{}, func_name, param_names);
}

File: java/epi/SearchInBst.java
package epi;

import generic_types.BstNode;
import test_framework.EpiTest;
import test_framework.GenericTest;

public class SearchInBst {

  public static BstNode<Integer> searchBST(BstNode<Integer> tree, int key) {
    return null;
  }

  @EpiTest(testDataFile = "../test_data/epi/search_in_bst.tsv")
  public static int searchBSTWrapper(BstNode<Integer> tree, int key) {
    BstNode<Integer> result = searchBST(tree, key);
    return result != null ? result.data : -1;
  }

  public static void main(String[] args) {
    System.exit(
        GenericTest
            .runFromAnnotations(args, "SearchInBst.java",
                new Object() {
                }.getClass().getEnclosingClass())
            .ordinal());
  }
}
